### PR TITLE
Fix for ogg trailing null bytes

### LINKED ISSUE #591: OGG Vorbis file save fails
Good day,
I am trying to save meta data to an Ogg Vorbis file that currently has no . I only have problems with this file. Unfortunately, I don't know where it is from. 
My code to write a tag to this file:
```python   
mutagen_file = mutagen.oggvorbis.OggVorbis("ogg_file")

mutagen_file["title"] = u"foo"
mutagen_file.save()
```

It throws the following error (incl. stack trace)
```
(stray-tagger-py3.10) ➜  stray-tagger git:(main) ✗  cd /home/christopher/Projects/stray-tagger
 ; /usr/bin/env /home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/bin/python /hom
e/christopher/.vscode-oss/extensions/ms-python.python-2022.20.2-universal/pythonFiles/lib/python/debugpy/adapter
/../../debugpy/launcher 49013 -- -m stray_tagger 
Traceback (most recent call last):
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/ogg.py", line 88, in __init__
    crc, segments) = struct.unpack("<4sBBqIIiB", header)
struct.error: unpack requires a buffer of 27 bytes

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/ogg.py", line 587, in save
    self.tags._inject(filething.fileobj, padding)
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/oggvorbis.py", line 147, in _inject
    OggPage.replace(fileobj, old_pages, new_pages)
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/ogg.py", line 436, in replace
    cls.renumber(fileobj, serial, sequence)
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/ogg.py", line 228, in renumber
    page = OggPage(fileobj)
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/ogg.py", line 90, in __init__
    raise error("unable to read full header; got %r" % header)
mutagen.ogg.error: unable to read full header; got b'\x00\x00'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/christopher/.vscode-oss/extensions/ms-python.python-2022.20.2-universal/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/__main__.py", line 39, in <module>
    cli.main()
  File "/home/christopher/.vscode-oss/extensions/ms-python.python-2022.20.2-universal/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 430, in main
    run()
  File "/home/christopher/.vscode-oss/extensions/ms-python.python-2022.20.2-universal/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 317, in run_module
    run_module_as_main(options.target, alter_argv=True)
  File "/home/christopher/.vscode-oss/extensions/ms-python.python-2022.20.2-universal/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 229, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/home/christopher/.vscode-oss/extensions/ms-python.python-2022.20.2-universal/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 186, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/home/christopher/.vscode-oss/extensions/ms-python.python-2022.20.2-universal/pythonFiles/lib/python/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 150, in _get_module_details
    __import__(pkg_name)
  File "/home/christopher/Projects/stray-tagger/stray_tagger/__init__.py", line 100, in <module>
    mutagen_file.save()
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/_util.py", line 160, in wrapper
    return func(self, h, *args, **kwargs)
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/ogg.py", line 589, in save
    reraise(self._Error, e, sys.exc_info()[2])
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/_util.py", line 47, in reraise
    raise tp(value).with_traceback(tb)
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/ogg.py", line 587, in save
    self.tags._inject(filething.fileobj, padding)
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/oggvorbis.py", line 147, in _inject
    OggPage.replace(fileobj, old_pages, new_pages)
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/ogg.py", line 436, in replace
    cls.renumber(fileobj, serial, sequence)
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/ogg.py", line 228, in renumber
    page = OggPage(fileobj)
  File "/home/christopher/.cache/pypoetry/virtualenvs/stray-tagger-O9q9BGp1-py3.10/lib/python3.10/site-packages/mutagen/ogg.py", line 90, in __init__
    raise error("unable to read full header; got %r" % header)
mutagen.oggvorbis.OggVorbisHeaderError: unable to read full header; got b'\x00\x00'
```

Before and after running the code above, I ran several ´oggz` commands to help with debugging:
Before running: 
```
$ oggz comment -l ogg_file       
Vorbis: serialno 0000000000
        Vendor: (null)
$ oggz validate ogg_file
$ oggz info ogg_file
Content-Duration: 00:03:46.771

Vorbis: serialno 0000000000
        31122 packets in 2274 pages, 13.7 packets/page, 1.129% Ogg overhead
        Audio-Samplerate: 44100 Hz
        Audio-Channels: 2
$ oggz codecs ogg_file
vorbis
$ 
```

After running:
```
$ oggz comment -l ogg_file       
Vorbis: serialno 0000000000
        Vendor: (null)
        title: foo
$ oggz validate ogg_file
$ oggz info ogg_file
Content-Duration: 00:03:24.886

Vorbis: serialno 0000000000
        25722 packets in 2034 pages, 12.6 packets/page, 1.114% Ogg overhead
        Audio-Samplerate: 44100 Hz
        Audio-Channels: 2
$ oggz codecs ogg_file
$ 
```
I tried to understand the code, but I was not able to find the mistake or bad section in my file, nor what generally happens. Maybe somebody knows what is wrong with my file.

I unfortunately cannot provide the (corrupt?) file due to copyright. Also, I don't have any other files than that. But maybe you can guide me through debugging this issue.

### PR DESCRIPTION
**Pull request by kism**
Fixes #591

So I have encountered some ogg files from some encoder that seems to append one or two null bytes to the file.

All existing tests pass.

### CODE REVIEW

**Comment by phw** [comment]
Seems useful. Is there some example file to test this? Would also be good to have tests covering this case.

**Comment by kism** [comment]
Found an encoding of a CC-BY-SA song that has the issue, Bad Sign by Brad Sucks
https://autist.network/adhoc/test.ogg
md5: `8e847801d93ecb8643f50a52e0e99f86`

```python
import mutagen
audio = mutagen.File("test.ogg")
audio['artist'] = "Brad Sucks"
audio.save()
```

Will cause

`mutagen.oggvorbis.OggVorbisHeaderError: unable to read full header; got b'\x00'`

In this case, the file has one null byte at the end, if I remove it with a hex editor and run that script, its happy.

If I just append null bytes to a good ogg file, then test with that script, it doesn't cause the error. So maybe it's todo with the length of the file per the ogg page headers.

File: mutagen/ogg.py [hunk 1]
```diff
@@ -81,6 +81,12 @@ def __init__(self, fileobj=None):
         self.offset = fileobj.tell()
 
         header = fileobj.read(27)
+
+        # If there is not enough data to make up the header...
+        # we might be looking at trailing null bytes on the file.
+        if len(header) != 27 and all(byte == 0 for byte in header):
+            header = b""
```
Comment (phw) [inline mutagen/ogg.py:88]: I think this should directly `raise EOFError`. This makes the code path clearer to read.

File: mutagen/ogg.py [hunk 2]
```diff
@@ -81,6 +81,12 @@ def __init__(self, fileobj=None):
         self.offset = fileobj.tell()
 
         header = fileobj.read(27)
+
+        # If there is not enough data to make up the header...
+        # we might be looking at trailing null bytes on the file.
+        if len(header) != 27 and all(byte == 0 for byte in header):
```
Comment (phw) [inline mutagen/ogg.py:87]: This works for most cases, but I see two potential issues here:

1. A file with exactly 27 trailing null bytes at the end will fail as before.
2. A file with 27 or more null bytes will also fail, but we don't know what comes after the 27 bytes. If it is only null bytes we could probably treat it as EOF as well. If there comes other garbage data failing seems correct.

Just wanted to mention this. Not sure how to handle this all. Probably your check is fine. This will solve the more common case of having only a few null bytes at the end. Everything above 26 null bytes we play save and fail as before.

I would just suggest to change the check to `len(header) < 27`. The result is the same, as only max. 27 bytes got read. But to me it makes the intent clearer.

**Review by phw — CHANGES_REQUESTED** [review]
Thanks for the example file. The fix mostly seems fine, with some minor comments.

But we really should have a test for this. Have a look at `test_ogg.py`, specifically the `test_renumber*` tests there. There is already a `test_renumber_extradata`, which tests failing for some garbage data at the end.

There could be a similar `test_renumber_trailing_null_bytes` that adds some null bytes (26, to test for the edge case) and checks that the call to `renumber` succeeded, but the trailing bytes got preserved.
